### PR TITLE
Add clarifications

### PR DESCRIPTION
**Pull request by claucece**
This clarifies the usage of Credential and Delegated Credential terms. It also clarifies the signing mechanism.

### CODE REVIEW

File: draft-ietf-tls-subcerts.md [hunk 1]
```diff
@@ -235,7 +236,7 @@ protocol is not allowed.
 Delegated credentials allow a peer to terminate TLS connections on behalf of
 the certificate owner.  If a credential is stolen, there is no mechanism for
 revoking it without revoking the certificate itself.  To limit exposure in case
-of delegated credential private key compromise, delegated credentials have a maximum
+of delegated credential's private key compromise, delegated credentials have a maximum
```
Comment (grittygrease) [inline draft-ietf-tls-subcerts.md:239]: How about "of the compomise of a delegated credential's private key, delegated credentials have a ..."

File: draft-ietf-tls-subcerts.md [hunk 2]
```diff
@@ -305,10 +306,10 @@ Client            Front-End            Back-End
   |        ...        |                    |
 ~~~~~~~~~~
 
-These two mechanisms can be complementary.  A server could use credentials for
-clients that support them, while using [KEYLESS] to support legacy clients.
+These two mechanisms can be complementary.  A server could use delegated credentials
+for clients that support them, while using [KEYLESS] to support legacy clients.
 The private key for a delegated credential can be used in place of a certificate
-private key, so it is important that the Front-End and Back-End are parties that
+private key, so it is important that the Front-End and Back-End are parties
```
Comment (grittygrease) [inline draft-ietf-tls-subcerts.md:312]: How about "that have" -> "with"

File: draft-ietf-tls-subcerts.md [hunk 3]
```diff
@@ -345,20 +346,20 @@ valid_time:
...
 expected_cert_verify_algorithm:
 
-: The signature algorithm of the credential key pair, where the type
+: The signature algorithm of the Credential key pair, where the type
   SignatureScheme is as defined in {{RFC8446}}. This is expected to be
   the same as CertificateVerify.algorithm sent by the server.  Only signature
-  algorithms allowed for use in CertificateVerify messages are allowed.  When
-  using RSA, the public key MUST NOT use the rsaEncryption OID, as a result,
+  algorithms allowed for use in Certificate Verify messages are allowed.  When
+  using RSA, the public key MUST NOT use the rsaEncryption OID. As a result,
```
Comment (grittygrease) [inline draft-ietf-tls-subcerts.md:353]: " " -> "  "

File: draft-ietf-tls-subcerts.md [hunk 4]
```diff
@@ -345,20 +346,20 @@ valid_time:
...
   rsa_pss_rsae_sha256, rsa_pss_rsae_sha384, rsa_pss_rsae_sha512.
 
 ASN1_subjectPublicKeyInfo:
 
-: The credential's public key, a DER-encoded {{X.690}} SubjectPublicKeyInfo as defined in
+: The Credential's public key, a DER-encoded {{X.690}} SubjectPublicKeyInfo as defined in
 {{RFC5280}}.
 
-The delegated credential has the following structure:
+The Delegated Credential has the following structure:
```
Comment (grittygrease) [inline draft-ietf-tls-subcerts.md:362]: For consistency, we should say "DelegatedCredential" instead.

File: draft-ietf-tls-subcerts.md [hunk 5]
```diff
@@ -612,8 +624,8 @@ stop these attacks due to the complexity of this mitigation [ROBOT].
 For TLS 1.2 servers that support RSA key exchange using a DC-enabled end-entity
 certificate, a hypothetical signature forgery attack would allow forging a
 signature over a delegated credential.
-The forged credential could then be used by the attacker as the equivalent of a
-man-in-the-middle certificate, valid for 7 days.
+The forged delegated credential could then be used by the attacker as the equivalent of a
+man-in-the-middle certificate, valid for maximum 7 days.
```
Comment (grittygrease) [inline draft-ietf-tls-subcerts.md:628]: "maximum" -> "a maximum of"

File: draft-ietf-tls-subcerts.md [hunk 6]
```diff
@@ -206,9 +207,9 @@ draft-02
 
 # Solution Overview
 
-A delegated credential is a digitally signed data structure with two semantic
+A delegated credential (DC) is a digitally signed data structure with two semantic
```
Comment (grittygrease) [inline draft-ietf-tls-subcerts.md:210]: Is the acryonym DC used anywhere except the diagram?

File: draft-ietf-tls-subcerts.md [hunk 7]
```diff
@@ -305,10 +306,10 @@ Client            Front-End            Back-End
...
 ~~~~~~~~~~
 
-These two mechanisms can be complementary.  A server could use credentials for
-clients that support them, while using [KEYLESS] to support legacy clients.
+These two mechanisms can be complementary.  A server could use delegated credentials
+for clients that support them, while using [KEYLESS] to support legacy clients.
 The private key for a delegated credential can be used in place of a certificate
-private key, so it is important that the Front-End and Back-End are parties that
+private key, so it is important that the Front-End and Back-End are parties
 have a trusted relationship.
```
Comment (chris-wood) [inline draft-ietf-tls-subcerts.md:313]:
```suggestion
private key, so it is important that the Front-End and Back-End are parties
with a trusted relationship.
```

File: draft-ietf-tls-subcerts.md [hunk 8]
```diff
@@ -345,20 +346,20 @@ valid_time:
 
 expected_cert_verify_algorithm:
 
-: The signature algorithm of the credential key pair, where the type
+: The signature algorithm of the Credential key pair, where the type
   SignatureScheme is as defined in {{RFC8446}}. This is expected to be
   the same as CertificateVerify.algorithm sent by the server.  Only signature
-  algorithms allowed for use in CertificateVerify messages are allowed.  When
-  using RSA, the public key MUST NOT use the rsaEncryption OID, as a result,
+  algorithms allowed for use in Certificate Verify messages are allowed.  When
```
Comment (chris-wood) [inline draft-ietf-tls-subcerts.md:352]:
```suggestion
  algorithms allowed for use in CertificateVerify messages are allowed.  When
```

File: draft-ietf-tls-subcerts.md (hunk 4 above)
Comment (chris-wood) [reply draft-ietf-tls-subcerts.md:362]: 👍

File: draft-ietf-tls-subcerts.md [hunk 9]
```diff
@@ -368,26 +369,34 @@ The delegated credential has the following structure:
...
 ~~~~~~~~~~
 
+cred:
+
+: The Credential structure as previously defined.
+
 algorithm:
 
-: The signature algorithm used to verify DelegatedCredential.signature.
+: The signature algorithm used to verify the DelegatedCredential.signature.
```
Comment (chris-wood) [inline draft-ietf-tls-subcerts.md:378]:
```suggestion
: The signature algorithm used to verify DelegatedCredential.signature.
```

File: draft-ietf-tls-subcerts.md [hunk 10]
```diff
@@ -368,26 +369,34 @@ The delegated credential has the following structure:
...
 
 signature:
 
 : The delegation, a signature that binds the credential to the end-entity
   certificate's public key as specified below. The signature scheme is specified
-  by DelegatedCredential.algorithm.
+  by the DelegatedCredential.algorithm.
```
Comment (chris-wood) [inline draft-ietf-tls-subcerts.md:384]:
```suggestion
  by DelegatedCredential.algorithm.
```

File: draft-ietf-tls-subcerts.md [hunk 11]
```diff
@@ -368,26 +369,34 @@ The delegated credential has the following structure:
...
+2. The context string "TLS, server delegated credentials" for server authentication and "TLS, client delegated credentials" for client authentication.
 3. A single 0 byte, which serves as the separator.
 4. The DER-encoded X.509 end-entity certificate used to sign the
    DelegatedCredential.
 5. DelegatedCredential.cred.
 6. DelegatedCredential.algorithm.
 
+The private key in the peer's end-entity certificate is used to sign this
+concatenation of values by using the algorithm indicated by DelegatedCredential.algorithm
+This creates the signature.
```
Comment (chris-wood) [inline draft-ietf-tls-subcerts.md:398]:
```suggestion
The signature is computed using the private key in the peer's end-entity certificate, with
the algorithm indicated by DelegatedCredential.algorithm.
```

File: draft-ietf-tls-subcerts.md [hunk 12]
```diff
@@ -435,7 +444,7 @@ CertificateEntry of its end-entity certificate; the client SHOULD ignore
 delegated credentials sent as extensions to any other certificate.
 
 The expected_cert_verify_algorithm field MUST be of a
-type advertised by the client in the SignatureSchemeList and is
+type advertised by the client in the SignatureSchemeList and it is
```
Comment (chris-wood) [inline draft-ietf-tls-subcerts.md:447]:
```suggestion
type advertised by the client in the SignatureSchemeList and is
```

File: draft-ietf-tls-subcerts.md [hunk 13]
```diff
@@ -481,7 +490,7 @@ peer's expected identity.  It also takes the following steps:
    certificate's notBefore value plus DelegatedCredential.cred.valid_time plus
    the maximum validity period.
 3. Verify that expected_cert_verify_algorithm matches
-   the scheme indicated in the peer's CertificateVerify message and that the
+   the scheme indicated in the peer's Certificate Verify message and that the
```
Comment (chris-wood) [inline draft-ietf-tls-subcerts.md:493]:
```suggestion
   the scheme indicated in the peer's CertificateVerify message and that the
```

File: draft-ietf-tls-subcerts.md (hunk 8 above)
Comment (chris-wood) [reply draft-ietf-tls-subcerts.md:352]: (`CertificateVerify` is the name of the handshake message type.)

File: draft-ietf-tls-subcerts.md [hunk 14]
```diff
@@ -491,9 +500,11 @@ peer's expected identity.  It also takes the following steps:
 
 If one or more of these checks fail, then the delegated credential is deemed
 invalid.  Clients and servers that receive invalid delegated credentials MUST terminate the
-connection with an "illegal_parameter" alert.  If successful, the participant
-receiving the Certificate message uses the public key in the credential to verify
-the signature in the peer's CertificateVerify message.
+connection with an "illegal_parameter" alert.
+
+If successful, the participant receiving the Certificate message uses the public
+key in the DelegatedCredential.cred to verify the signature in the peer's
```
Comment (chris-wood) [inline draft-ietf-tls-subcerts.md:506]:
```suggestion
key in DelegatedCredential.cred to verify the signature in the peer's
```

File: draft-ietf-tls-subcerts.md [hunk 15]
```diff
@@ -491,9 +500,11 @@ peer's expected identity.  It also takes the following steps:
...
+Certificate Verify message.
```
Comment (chris-wood) [inline draft-ietf-tls-subcerts.md:507]:
```suggestion
CertificateVerify message.
```

File: draft-ietf-tls-subcerts.md (hunk 5 above)
Comment (chris-wood) [inline draft-ietf-tls-subcerts.md:628]:
```suggestion
man-in-the-middle certificate, valid for a maximum of 7 days.
```

File: draft-ietf-tls-subcerts.md (hunk 6 above)
Comment (claucece) [reply draft-ietf-tls-subcerts.md:210]: Yes! Sometimes, for example, on `DC-enabled`, `non-DC traffic`.

**Comment by claucece** [comment]
@grittygrease @chris-wood I've incorporated the comments ;)

**Pull request merged**